bin/add_celltypes_to_sce.R

```diff
@@ -250,6 +267,47 @@ if (!is.null(opt$cellassign_predictions)) {
     }
     metadata(sce)$cellassign_reference_organs <- cellassign_organs
   }
+} else {
+  has_cellassign <- FALSE # use to decide if we should assign consensus labels
+}
+
+# assign consensus cell type labels
+if (has_singler & has_cellassign) {
+  # now make sure that reference file exists
+  stopifnot(
+    "Consensus cell type reference file does not exist" = file.exists(opt$consensus_celltype_ref)
+  )
+
+  # read in consensus table
+  consensus_ref_df <- readr::read_tsv(opt$consensus_celltype_ref) |>
+    # select unique combinations of consensus refs based on ontology columns
+    dplyr::select(blueprint_ontology, panglao_ontology, consensus_ontology, consensus_annotation) |>
+    unique()
```

Comment:
For uniformity (and a touch faster?)
```suggestion
    dplyr::distinct()
```